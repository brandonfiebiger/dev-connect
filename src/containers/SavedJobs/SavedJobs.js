import React, { Component } from 'react';
import SavedCard from '../SavedCard/SavedCard';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeJob } from '../../actions';

export class SavedJobs extends Component {
  displaySavedCard = () => {
    const { savedJobs, jobs, jobTypes } = this.props;

    if (!savedJobs.length) {
      return;
    }
    return savedJobs.map(id => {
      const matchedJob = jobs.find(job => job.id === id);
      const matchedJobType = jobTypes.find(
        type => type.id === matchedJob.job_title_id
      );

      return (
        <SavedCard
          description={matchedJob.description}
          company={matchedJob.company}
          location={matchedJob.location}
          title={matchedJobType.job_title}
          status={matchedJob.status}
          id={matchedJob.id}
        />
      );
    });
  };

  render() {
    const savedCards = this.displaySavedCard();
    return (
      <div>
        <h1>Saved Jobs</h1>
        <NavLink className="jobs-container-link" to="/">
          Home
        </NavLink>
        {savedCards}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  jobs: state.jobs,
  jobTypes: state.jobTypes,
  savedJobs: state.savedJobs
});

export const mapDispatchToProps = dispatch => ({
  removeJob: job => dispatch(removeJob(job))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SavedJobs);
