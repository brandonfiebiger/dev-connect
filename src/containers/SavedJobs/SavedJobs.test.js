import React from 'react';
import { shallow } from 'enzyme';
import { SavedJobs } from './SavedJobs';

describe('SavedJobs', () => {
  let wrapper;
  let mockSaved = [{ 0: 52 }, { 1: 63 }, { 2: 67 }];
  let mockJobs = [
    {
      id: 121,
      description: 'A good job',
      company: 'DHI GROUP, INC.',
      location: 'GREATER DENVER AREA',
      status: 'none',
      job_title_id: 1
    },
    {
      id: 122,
      description: 'A good job',
      company: 'TRAVELERS HAVEN',
      location: 'GREATER DENVER AREA',
      status: 'saved',
      job_title_id: 2
    }
  ];
  let mockTypes = [
    {
      id: 1,
      job_title: 'Application Architect',
      average_salary: 85000
    },
    {
      id: 2,
      job_title: 'Software Engineer',
      average_salary: 75000
    }
  ];

  beforeEach(() => {
    wrapper = shallow(
      <SavedJobs
        jobs={mockJobs}
        jobTypes={mockTypes}
        savedJobs={mockSaved}
        removeJob={jest.fn()}
      />
    );
  });

  it.skip('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
