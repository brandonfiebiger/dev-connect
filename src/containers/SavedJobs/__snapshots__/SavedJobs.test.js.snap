// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedJobs should match the snapshot 1`] = `undefined`;
